### PR TITLE
AC-988 : Add backslash in AddEditPatientFragment

### PR DESCRIPTION
**Pull request by rdodiya**
## Description of what I changed

In Patient Registration, add backslash in DOB editText dynamically and  add 0 before date & month while selecting date from datepicker to maintain date format.

## Issue I worked on

JIRA Issue: https://issues.openmrs.org/browse/AC-988

## Test app after changes
ac988

https://user-images.githubusercontent.com/75332758/139407879-8207b13f-4c39-417e-9352-aa59c9e903df.mp4

## Checklist: I completed these to help reviewers :)

- [x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

- [ ] I have **added tests** to cover my changes. (If you refactored
existing code that was well tested you do not have to add tests)

- [x] All new and existing **tests passed**.

- [x] My pull request is **based on the latest changes** of the master branch.

### CODE REVIEW

**Comment by LuGO0** [comment]
can you post a previous Image as well just for history.

**Comment by rdodiya** [comment]
> can you post a previous Image as well just for history. thanks !!

https://issues.openmrs.org/secure/attachment/60106/Screenshot_20210331-175757_OpenMRS_Android_Client%5B1%5D.jpg

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/addeditpatient/AddEditPatientFragment.java [hunk 1]
```diff
@@ -568,6 +568,20 @@ public void beforeTextChanged(CharSequence s, int start, int count, int after) {
...
+                        binding.dobEditText .setText(new StringBuilder(binding.dobEditText .getText().toString()).insert(str.length() - 1, "/").toString());
+                        binding.dobEditText .setSelection(binding.dobEditText .getText().length());
+                    }
+                }if (textLength == 6) {
+                    if (!str.substring(3).contains("/")) {
+                        binding.dobEditText .setText(new StringBuilder(binding.dobEditText .getText().toString()).insert(str.length() - 1, "/").toString());
+                        binding.dobEditText .setSelection(binding.dobEditText .getText().length());
+                    }
+                }
             }
```
Comment (LuGO0) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/addeditpatient/AddEditPatientFragment.java:585]: can you reformat the change?

File: openmrs-client/src/main/res/layout/fragment_patient_info.xml [hunk 2]
```diff
@@ -371,7 +371,8 @@
                                     android:inputType="datetime"
                                     android:maxLines="1"
                                     android:nextFocusDown="@+id/estimatedYear"
-                                    android:textSize="14sp" />
+                                    android:textSize="14sp"
+                                    android:maxLength="10"/>
                             </com.google.android.material.textfield.TextInputLayout>
```
Comment (LuGO0) [inline openmrs-client/src/main/res/layout/fragment_patient_info.xml:376]: do we need a max length restriction?

**Review by LuGO0 — CHANGES_REQUESTED** [review]

**Comment by rdodiya** [comment]
@LuGO0  Please inform if there is more to reformat in code.

**Comment by LuGO0** [comment]
can you squash these changes @rdodiya ?

**Comment by rdodiya** [comment]
I am creating new PR ,there is some problem i am faced after squashed the commit

**Pull request closed without merging**